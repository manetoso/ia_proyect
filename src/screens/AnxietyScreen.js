import React, { useState } from 'react'
import { Redirect } from 'react-router';
import Swal from "sweetalert2";

import QuestionnaireAnx from "../components/questionaire_anx";
import { QuestionsAnx } from "../components/Questions";
import Fuzzy from "../middlewares";

export const AnxietyScreen = () => {

    const [values, setvalues] = useState({});
    const [results, setResults] = useState(false)
    const [cancelQuestionnaire, setCancelQuestionnaire] = useState(false)

    const getValues = (target, value) => {
        setvalues((values) => ({ ...values, [target]: value }));
    };

    const hadleCancel = () => {
        Swal.fire({
        title: "¿Está seguro de cancelar el test?",
        text: 'Se le enviará al menú de Inicio',
        icon: "warning",
        showCancelButton: true,
        confirmButtonText: "Si",
        cancelButtonText: "No",
        confirmButtonColor: "#21445F",
        cancelButtonColor: "#607D8B",
        }).then( ( result ) => {
            if ( result.isConfirmed ) {
                setCancelQuestionnaire( true )
            } else if ( result.dismiss === Swal.DismissReason.cancel ) {
                Swal.fire({
                    icon: 'success',
                    title: 'Entendido',
                    text: 'Siga contestando su test',
                    timer: 2000,
                    timerProgressBar: true,
                    showConfirmButton: false,
                })
            }
        })
    };

    const handleSubmit = () => {
        if ( Object.keys( values ).length !== 10 ) {
        return Swal.fire('Oops...', 'Por favor, responda todas las preguntas antes de terminar', 'error')
        }
        setResults( !results )
    }

    if ( cancelQuestionnaire ) {
        return <Redirect to="/" />
    }

    return (
        <>
            <div className="container mb-5">
            
                {
                    results
                    ? <Fuzzy data={ values } />
                    : QuestionsAnx.map((question, key) => (
                        <QuestionnaireAnx
                        key={key}
                        number={key + 1}
                        action={getValues}
                        question={question}
                        />
                    ))
                }

                <div className="d-flex justify-content-end">
                    <button
                        type="button"
                        style={{ backgroundColor: "#607D8B" }}
                        className="btn btn-secondary me-3"
                        onClick={ hadleCancel }
                    >
                        Cancelar
                    </button>
                    <button
                        type="button"
                        style={{ backgroundColor: "#607D8B" }}
                        className="btn btn-secondary"
                        onClick={ handleSubmit }
                    >
                        Enviar
                    </button>
                </div>
            </div>
        </>
    )
}
