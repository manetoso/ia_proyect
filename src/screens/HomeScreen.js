import React from 'react'
// import PropTypes from 'prop-types'

function HomeScreen(props) {
    return (
        <div>
            <h1>Home</h1>
        </div>
    )
}

HomeScreen.propTypes = {

}

export default HomeScreen

