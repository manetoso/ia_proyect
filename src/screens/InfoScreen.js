import React from 'react'
import portada2 from "../images/t2.jpg"

export const InfoScreen = () => {
    return (
        <div className="container" >
            <h1 className="text-center">Atención psicológica</h1>
            <div className="mt-3">
                <div className="bg-primary pt-3 px-3 pt-md-5 px-md-5 text-center text-white overflow-hidden">
                    <div className="my-3 py-3">
                        <h4>Si necesitas atención psicológica gratuita y no sabes a donde acudir, aquí encontrarás alternativas útiles para poder atenderte. Hay organizaciones con psicólogos muy cualificados, que brindan apoyo psicológico gratuito y de calidad en casos de emergencia. </h4>
                    </div>
                    <div className="bg-light shadow-sm mx-auto" 
                        style={{width: "80%", height: "500px", borderRadius: "21px 21px 0 0"}}>
                        <img src={portada2} className="d-block w-100" alt="..." />
                    </div>
                </div>
            </div>
            <div className="mt-3">
                <div className="row row-cols-1 row-cols-md-2 g-4" >
                    <div className="col">
                        <div className="shadow-lg p-3 mb-5 bg-info rounded" 
                             style={{ height: "20rem" }}>
                            <div className="card text-white bg-primary mb-3" 
                                 style={{ height: "18rem" }}>
                                <div className="card-body text-center">
                                    <h3 className="card-title">Linares García Ana Paola</h3>
                                    <h4 className="card-title">Lic. en psicología clínica</h4>
                                    <hr></hr>
                                    <h5>Solicitar cita por correo:</h5>
                                    <a href="https://www.gmail.com"><ul>paolinares95@gmail.com</ul></a>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div className="col">
                        <div className="shadow-lg p-3 mb-5 bg-info rounded" 
                             style={{ height: "20rem" }}>
                            <div className="card text-white bg-primary mb-3" 
                                 style={{ height: "18rem" }}>
                                <div className="card-body text-center">
                                    <h3 className="card-title">Maldonado Rodríguez Iliana Cecilia</h3>
                                    <h3 className="card-title">Lic. en psicología clínica</h3>
                                    <hr></hr>
                                    <h5>Solicitar cita por correo:</h5>
                                    <a href="https://www.gmail.com"><ul>ilianarodriguez96@gmail.com</ul></a>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div className="col">
                        <div className="shadow-lg p-3 mb-5 bg-info rounded" style={{ height: "20rem" }}>
                            <div className="card text-white bg-primary mb-3" style={{ height: "18rem" }}>                            
                                <div className="card-body text-center">
                                    <h3 className="card-title">García Muñiz Areli Ariana</h3>
                                    <h3 className="card-title">Lic. en psicología clínica</h3>
                                    <hr></hr>
                                    <h5>Solicitar cita por correo:</h5>
                                    <a href="https://www.gmail.com"><ul>arimuniz98@gmail.com</ul></a>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div className="col">
                        <div className="shadow-lg p-3 mb-5 bg-info rounded" style={{ height: "20rem" }}>
                            <div className="card text-white bg-primary mb-3" style={{ height: "18rem" }}>                            
                                <div className="card-body text-center">
                                    <h3 className="card-title">Rodríguez Almanza Francisco Salvador</h3>
                                    <h3 className="card-title">Lic. en psicología educativa</h3>
                                    <hr></hr>
                                    <h5>Solicitar cita por correo:</h5>
                                    <a href="https://www.gmail.com"><ul>paco94@gmail.com</ul></a>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
            <h3>La asistencia telefónica es el recurso más utilizado para urgencias, por lo que los números 800 son gratis en todo México.</h3>
            <div className="row row-cols-1 row-cols-md-2 g-4" >
                <div className="col">
                    <div className="shadow-lg p-3 mb-5 bg-info rounded" style={{ height: "21rem" }}>
                        <div className="card text-white bg-primary mb-3" style={{ height: "19rem" }}>                           
                            <div className="card-body text-center">
                                <h3 className="card-title">SIPINNA</h3>
                                <h5>Personal especializado en atención de niñas, niños y adolecentes</h5>
                                <hr></hr>
                                <h5>Linea de Atención telefónica</h5>
                                <ul>800 343 48 61</ul>
                                <h5>Atención por Facebook</h5>
                                <a href="https://www.facebook.com/SIPINNAGTO/"><ul>@SIPINNAGTO</ul></a>
                            </div>
                        </div>
                    </div>
                </div>

                <div className="col">
                    <div className="shadow-lg p-3 mb-5 bg-info rounded"  style={{ height: "21rem" }}>
                        <div  className="card text-white bg-primary mb-3" style={{ height: "19rem" }}>                            
                            <div className="card-body text-center">
                                <h3 className="card-title">Dinámicamente</h3>
                                <h5>Personal especializado de psicología está 24 horas del día 365 días del año</h5>
                                <hr></hr>
                                <h5>Linea de Atención telefónica</h5>
                                <ul>800 290 00 24</ul>
                                <h5>Atención por página web</h5>
                                <a href="https://www.dinamicamente.mx">dinamicamente.mx</a>
                            </div>
                        </div>
                    </div>
                </div>

                <div className="col">
                    <div className="shadow-lg p-3 mb-5 bg-info rounded"  style={{ height: "20rem" }}>
                        <div className="card text-white bg-primary mb-3" style={{ height: "18rem" }}>
                            <div className="card-body text-center">
                                <h3 className="card-title">Línea de intervención</h3>
                                <hr></hr>
                                <h5>Línea de intervención en crisis psicológica</h5>
                                <ul>800 224 24 80</ul>
                            </div>
                        </div>
                    </div>
                </div>

                <div className="col">
                    <div className="shadow-lg p-3 mb-5 bg-info rounded"  style={{ height: "20rem" }}>
                        <div className="card text-white bg-primary mb-3" style={{ height: "18rem" }}>                            
                            <div className="card-body text-center">
                                <h3 className="card-title">Centro de Atención Psicológico Integral</h3>
                                <hr></hr>
                                <h5>Linea de Atención telefónica</h5>
                                <ul>461 100 9038</ul>
                            </div>
                        </div>
                    </div>
                </div>
            </div>


        </div>
    )
}
