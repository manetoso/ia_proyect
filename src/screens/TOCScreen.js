import React, { useState } from "react";
import { Redirect } from "react-router";
import Swal from "sweetalert2";
import { useHistory } from 'react-router-dom';
import QuestionnaireAnx from "../components/questionaire_anx";
import { QuestionsTOC } from "../components/Questions";
import Fuzzy from "../middlewares";

export const TOCScreen = () => {
  
  let history =  useHistory()

  const [values, setvalues] = useState({});
  const [results, setResults] = useState(false);
  const [cancelQuestionnaire, setCancelQuestionnaire] = useState(false);

  const getValues = (target, value) => {
    setvalues((values) => ({ ...values, [target]: value }));
  };

  const handleInfo = () => {
    Swal.fire({
        title: "Se te Redigirá a una ventana con Información",
        text: 'Obtendras información de contacto C:',
        icon: "info",
        showCancelButton: true,
        confirmButtonText: "Si",
        cancelButtonText: "No",
        confirmButtonColor: "#21445F",
        cancelButtonColor: "#607D8B",
        }).then( (result  )=>{
            if( result.isConfirmed ) {
                history.push("/info")
            }
            else if ( result.dismiss === Swal.DismissReason.cancel ) {
                Swal.fire({
                    icon: 'success',
                    title: 'Entendido',
                    text: 'Siga contestando su test',
                    timer: 2000,
                    timerProgressBar: true,
                    showConfirmButton: false,
                })
            }
        })
  }

  const hadleCancel = () => {
    Swal.fire({
      title: "¿Está seguro de cancelar el test?",
      text: "Se le enviará al menú de Inicio",
      icon: "warning",
      showCancelButton: true,
      confirmButtonText: "Si",
      cancelButtonText: "No",
      confirmButtonColor: "#21445F",
      cancelButtonColor: "#607D8B",
    }).then((result) => {
      if (result.isConfirmed) {
        setCancelQuestionnaire(true);
      } else if (result.dismiss === Swal.DismissReason.cancel) {
        Swal.fire({
          icon: "success",
          title: "Entendido",
          text: "Siga contestando su test",
          timer: 2000,
          timerProgressBar: true,
          showConfirmButton: false,
        });
      }
    });
  };

  const handleSubmit = () => {
    if (Object.keys(values).length !== 10) {
      return Swal.fire(
        "Oops...",
        "Por favor, responda todas las preguntas antes de terminar",
        "error"
      );
    }
    setResults(!results);
  };

  if (cancelQuestionnaire) {
    return <Redirect to="/" />;
  }
  return (
    <>
      <div className="container mb-5">
        <h1>Trastorno Obsesivo Compulsivo ( TOC )</h1>
        {
        results ? 
          (
            <Fuzzy data={values} topic="TOC" />
          ) 
          : ([
            <h3 key="01">Indique su respuesta de acuerdo a los siguientes valores:</h3>,
            <p className="lead" key="02">5 - Siempre , 4 - Casi siempre, 3 - En ocasiones, 2 - Casi nunca, 1 - Nunca </p>,
            QuestionsTOC.map((question, key) => (
              <QuestionnaireAnx
                key={key}
                number={key + 1}
                action={getValues}
                question={question}
              />
            ))
            ])
        }
        {
          results
          ? (
              <div className="d-flex justify-content-end">
                  <button 
                  id="btn-infor"
                  type="button"
                  style={{ backgroundColor: "#607D8B" }}
                  className="btn btn-secondary me-3"
                  onClick={ handleInfo }
                  > Mostrar Informacion de Contacto</button>
              </div>
          )
          :( <div className="d-flex justify-content-end">
              <button
                  id="btn-cancel"
                  type="button"
                  style={{ backgroundColor: "#607D8B" }}
                  className="btn btn-secondary me-3"
                  onClick={ hadleCancel }
              >
                  Cancelar
              </button>
              <button
                  id="btn-send"
                  type="button"
                  style={{ backgroundColor: "#607D8B" }}
                  className="btn btn-secondary"
                  onClick={ handleSubmit }
              >
                  Enviar
              </button>
          </div>)
        }        
      </div>
    </>
  );
};
