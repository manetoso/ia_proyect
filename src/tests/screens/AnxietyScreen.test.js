import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from "enzyme"

import { AnxietyScreen } from '../../screens/AnxietyScreen';

//LINK DE LA DOCUMENTACIÓN EN ENZYME CON SHALLOW https://enzymejs.github.io/enzyme/docs/api/shallow.html

//SE CREA EL DOCUMENTO DE PRUEBAS
describe('Testing in <AnxietyScreen />', () => {
    
    //VARIABLES QUE LLENAN LOS CAMPOS NECESARIOS PARA REALIZAR PRUEBAS EN EL COMPONENTE
    

    //INTICA QUE EL WRAPPER ES EL CONTENIDO HTML DEL COMPONENTE
    const wrapper = shallow( <AnxietyScreen /> )

    //TODOS LOS TESTS SON PRUEBAS UNITARIAS DENTRO DE ESTE DOCUMENTO DE PRUEBAS

    //TEST QUE BUSCA QUE SE MUESTRE CORRECTAMENTE EL CONTENIDO HTML DEL COMPONENTE
    test('should show the screen correctly', () => {
        //ESCRIBE UN NUEVO SNAPTSHOT EN LA CARPETA DE _snapshots_
        //DENTRO DEL DIRECTORIO DE PRUEBAS DEL COMPONENTE AL EJECUTARCE POR PRIMERA VEZ

        //SI EXISTE UNA SNAPTSHOT LA USA PARA COMPARAR EL WRAPPER CON ESTA
        //SI SON IGUALES TODO ESTÁ BIEN, SI NO SE PARECEN EN ALGO LA PRUEBA FALLA
        expect( wrapper ).toMatchSnapshot()
    })

    //TEST QUE BUSCA QUE SE MUESTREN UN TOTAL DE 10 PREGUNTAS EN LA PÁGINA
    test('should show a total of 10 questions', () => {
        const questionsLength = wrapper.find( 'QuestionnaireAnx' )

        expect( questionsLength ).toHaveLength( 10 )
    })

    //TEST QUE BUSCA QUE SE MUESTRE UN BOTÓN PARA CANCELAR
    test('should show a button to cancel', () => {
        const cancelButton = wrapper.find( '#btn-cancel' )

        expect( cancelButton ).toHaveLength( 1 )
    })

    //TEST QUE BUSCA QUE SE MUESTRE UN BOTÓN PARA TERMINAR EL TEST
    test('should show a button to finish the form', () => {
        const sendButton = wrapper.find( '#btn-send' )

        expect( sendButton ).toHaveLength( 1 )
    })

})
