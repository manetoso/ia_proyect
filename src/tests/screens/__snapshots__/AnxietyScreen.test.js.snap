// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <AnxietyScreen /> should show the screen correctly 1`] = `
<Fragment>
  <div
    className="container mb-5"
  >
    <QuestionnaireAnx
      action={[Function]}
      key="0"
      number={1}
      question="¿Te llegas a sentir de forma muy incómoda en situaciones sociales
    como en conversaciones, mientras estas comiendo/bebiendo o dar 
    alguna charla por miedo a ser examinado o juzgado?"
    />
    <QuestionnaireAnx
      action={[Function]}
      key="1"
      number={2}
      question="¿Te cohíbes en tu forma de ser o no actúas de cierta manera para 
    que no te humillen o recibas algún tipo de rechazo social?"
    />
    <QuestionnaireAnx
      action={[Function]}
      key="2"
      number={3}
      question="Ahora piensa que estas en una reunión con tus amigos, pero 
    también con personas que no conoces ¿Sientes incomodidad o 
    miedo para expresarte o comportarte como normalmente lo haces?"
    />
    <QuestionnaireAnx
      action={[Function]}
      key="3"
      number={4}
      question="Con frecuencia evito situaciones sociales ya que estas me llegan a 
    causar incomodidad o hasta miedo."
    />
    <QuestionnaireAnx
      action={[Function]}
      key="4"
      number={5}
      question="Es tanto mi miedo a las situaciones sociales a tal grado que no solo 
    me afecta en lo social, también me afecta en el campo laboral."
    />
    <QuestionnaireAnx
      action={[Function]}
      key="5"
      number={6}
      question="¿Cuándo estoy en situaciones donde tengo que involucrarme 
    socialmente llego a padecer algún síntoma de ansiedad como por 
    ejemplo palpitaciones, temblores, molestias gastrointestinales o 
    tensión muscular?"
    />
    <QuestionnaireAnx
      action={[Function]}
      key="6"
      number={7}
      question="¿Esta ansiedad o miedo a las situaciones sociales a durado más de 
    6 meses?"
    />
    <QuestionnaireAnx
      action={[Function]}
      key="7"
      number={8}
      question="Evito socializar o situaciones sociales porque padezco una 
    disformidad corporal u algún tipo de problema de salud."
    />
    <QuestionnaireAnx
      action={[Function]}
      key="8"
      number={9}
      question="Evito socializar o situaciones sociales porque padezco una 
    disformidad corporal u algún tipo de problema de salud."
    />
    <QuestionnaireAnx
      action={[Function]}
      key="9"
      number={10}
      question="La ansiedad o miedo que siento, no solo me ocurre en 
    situaciones donde se me involucre socialmente."
    />
    <div
      className="d-flex justify-content-end"
    >
      <button
        className="btn btn-secondary me-3"
        id="btn-cancel"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#607D8B",
          }
        }
        type="button"
      >
        Cancelar
      </button>
      <button
        className="btn btn-secondary"
        id="btn-send"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#607D8B",
          }
        }
        type="button"
      >
        Enviar
      </button>
    </div>
  </div>
</Fragment>
`;
